tools/imagepuller-eval/main.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"maps"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/edgelesssys/contrast/imagepuller/client"
+	"github.com/shirou/gopsutil/v3/disk"
+	"github.com/spf13/cobra"
+)
+
+var imageList = []string{
+	"ghcr.io/edgelesssys/contrast/dmesg:v0.0.1@sha256:6ad6bbb5735b84b10af42d2441e8d686b1d9a6cbf096b53842711ef5ddabd28d",
+	"ghcr.io/charludo/contrast/coordinator@sha256:6f966a922cc9a39d7047ed41ffafc7eb7a3c6a4fd8966cbf30fa902b455789f7",
+	"tensorflow/tensorflow:latest-gpu@sha256:73fe35b67dad5fa5ab0824ed7efeb586820317566a705dff76142f8949ffcaff",
+}
+var mountPoint = "current_server"
+
+func getDiskUsage(path string) (uint64, error) {
+	usage, err := disk.Usage(path)
+	if err != nil {
+		return 0, err
+	}
+	return usage.Used, nil
+}
+
+func extractName(name string) string {
+	at := strings.Index(name, "@")
+	if at == -1 {
+		return ""
+	}
+	slash := strings.LastIndex(name[:at], "/")
+	return name[slash+1 : at]
+}
+
+func cleanup(storagePath string) {
+	cmd := exec.Command("findmnt", "-rn", "-o", "TARGET")
+	output, err := cmd.Output()
+	if err != nil {
+		log.Fatalf("failed to run findmnt: %v", err)
+	}
+
+	var mountpoints []string
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, storagePath) {
+			mountpoints = append(mountpoints, line)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("error reading findmnt output: %v", err)
```

Comment:
Fatalf should only be used in a top-level main/run function. Return an error instead.